### ISSUE TITLE
BUG - odata-service-writer overwrites existing service in ui5*.yaml when a 2nd service is added

### ISSUE THREAD

**Issue by tobiasqueck** [SAP/open-ux-tools#279]
### Description
The `odata-service-writer` ignores existing services in a project and overwrites them if a 2nd service is to be added.

### Steps to Reproduce
1. Add the following test case to `test/unit/index.test.ts`
```
it('Add a valid 2nd service', async () => {
        const config = {
            ...commonConfig,
            version: OdataVersion.v2
        } as OdataService;
        await generate(testDir, config, fs);

        const config2 = {
            ...commonConfig,
            version: OdataVersion.v4,
            path: '/v4/odata/another'
        } as OdataService;
        await generate(testDir, config2, fs);

        expect(fs.read(join(testDir, 'ui5.yaml'))).toMatchInlineSnapshot();
        expect(fs.read(join(testDir, 'ui5-mock.yaml'))).toMatchInlineSnapshot();
    });
```
2. Check the newly generated snapshot in the console
3. See error

### Expected results
`ui5.yaml` contains two entries for `backend`
`ui5-mock.yaml` contains a valid configuration for both services

### Actual results
The entries in both yaml files are only reflecting the 2nd service.

### Suggested Solution
Fixing this problem requires multiple different fixes:
1. `ui5-config` needs an API for CRUD operations on middlewares (and tasks - not required here but should be considered to be done together), so that consumers can check if there are existing entries and create/update accordingly
2. `odata-service-writer` needs to use the new API to update the proxy and mock middlewares if they are already configured
3. If more than one service is to be mocked, then the configuration needs to be extracted into a separate mockserver configuration file

#### Version/Components/Environment	
Add any other context about the problem here
OS:                      
- [x] Mac OS
- [x] Windows

### Problem
{describe the problem} 

### Fix
{describe the fix} 

### Why was it missed
This use case was not part of the initially tested use cases of the `odata-service-writer` since it is not covered in the SAP Fiori generator.

### How can we avoid this
It can be discussed whether this is a bug report or a feature request since the scenario was not defined as a clear requirement.

**Comment by cfg74** [comment]
Thanks for heads-up @tobiasqueck, we have to add multi-service capabilities. Question, the config writer also adds the dev dependency to the mockserver in the package.json, but doesn't issue an npm install, would this conceptually belong to the command's scope or not?

**Comment by cfg74** [comment]
@tobiasqueck second point, the issue mentions the CRUD approach. I'm wondering if this is required for mockserver config handling. AFAIK the config writer reads the manifest information about metadata, annotations and their local copy paths + the ui5.yaml and creates a ui5-mock.yaml from it. Wouldn't it be sufficient to recreate the ui5-mock.yaml and only preserve the customer's mockserver settings, as the rest is not original information?

**Comment by tobiasqueck** [comment]
> Thanks for heads-up @tobiasqueck, we have to add multi-service capabilities. Question, the config writer also adds the dev dependency to the mockserver in the package.json, but doesn't issue an npm install, would this conceptually belong to the command's scope or not?

Yes, that would belong to the command. Since the config write doesn't know whether it is called for an existing project or during the project generation. I had a quick look and create a rudimentary `add service` command and noticed that it actually works, as long as you provide a specific service and model name.
Seems like the only bug is in the mockserver config generation.

**Comment by tobiasqueck** [comment]
> @tobiasqueck second point, the issue mentions the CRUD approach. I'm wondering if this is required for mockserver config handling. AFAIK the config writer reads the manifest information about metadata, annotations and their local copy paths + the ui5.yaml and creates a ui5-mock.yaml from it. Wouldn't it be sufficient to recreate the ui5-mock.yaml and only preserve the customer's mockserver settings, as the rest is not original information?

Agree, the command should just ensure that `manifest.json` and `ui5-mock.yaml` are in sync.
